src/ServiceControl/Operations/UnitOfWork/InMemory/InMemoryIngestionUnitOfWork.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace ServiceControl.Operations
+{
+    using System.Collections.Concurrent;
+    using System.Threading.Tasks;
+    using Monitoring;
+
+    class InMemoryIngestionUnitOfWork : IIngestionUnitOfWork
+    {
+        ConcurrentBag<KnownEndpoint> knownEndpoints = new ConcurrentBag<KnownEndpoint>();
+        IMonitoringDataStore dataStore;
+
+        public InMemoryIngestionUnitOfWork(IMonitoringDataStore dataStore)
+        {
+            this.dataStore = dataStore;
+            Monitoring = new InMemoryMonitoringIngestionUnitOfWork(this);
+        }
+
+        public IMonitoringIngestionUnitOfWork Monitoring { get; }
+        public IRecoverabilityIngestionUnitOfWork Recoverability { get; }
+
+        public async Task Complete()
+        {
+            foreach (var endpoint in knownEndpoints)
+            {
+                await dataStore.CreateIfNotExists(endpoint.EndpointDetails)
```

Comment:
Not related to this change, but this `CreateIfNotExists` is very database-centric and does not sound correct. Should it be named just 'store' instead?

```suggestion
                await dataStore.CreateIfNotExists(endpoint.EndpointDetails)
```

Reply:
In this case it definitely should not update an existing instance. We have `CreateIfNotExists` and `CreateOrUpdate`.